### PR TITLE
[storage-preview] Support Blob inventory

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
storage-preview

File: src/storage-preview/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
  Release History
```
Comment (evelyn-ys) [inline src/storage-preview/HISTORY.rst:3]:
```suggestion
Release History
```

File: src/storage-preview/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
...
 ===============
+0.6.0 (2020-11-04)
+++++++++++++++++
```
Comment (evelyn-ys) [inline src/storage-preview/HISTORY.rst:6]:
```suggestion
++++++++++++++++++
```

File: src/storage-preview/HISTORY.rst [hunk 3]
```diff
@@ -2,6 +2,10 @@
...
+* Support Blob Inventory Policy in storage account
+
 0.5.0 (2020-11-02)
 ++++++++++++++++
```
Comment (evelyn-ys) [inline src/storage-preview/HISTORY.rst:10]:
```suggestion
++++++++++++++++++
```

**Review by evelyn-ys — COMMENTED** [review]
some markdown formatting issues

File: src/storage-preview/azext_storage_preview/_params.py [hunk 4]
```diff
@@ -69,6 +71,42 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('if_match')
         c.argument('if_none_match')
 
+    with self.argument_context('storage account blob-inventory-policy') as c:
+        c.ignore('blob_inventory_policy_name')
+        c.argument('resource_group_name', required=False, validator=process_resource_group)
+        c.argument('account_name',
+                   help='The name of the storage account within the specified resource group. Storage account names '
+                        'must be between 3 and 24 characters in length and use numbers and lower-case letters only.')
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:79]: Why not reuse `acct_name_type`

Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/_params.py:79]: option `-n` is not suitlable

File: src/storage-preview/azext_storage_preview/operations/account.py [hunk 5]
```diff
@@ -143,6 +143,60 @@ def update_storage_account(cmd, instance, sku=None, tags=None, custom_domain=Non
...
+    if os.path.exists(policy):
+        policy = get_file_json(policy)
+    else:
+        policy = shell_safe_json_parse(policy)
+
+    BlobInventoryPolicy, InventoryRuleType, BlobInventoryPolicyName = \
+        cmd.get_models('BlobInventoryPolicy', 'InventoryRuleType', 'BlobInventoryPolicyName')
+    properties = BlobInventoryPolicy()
+    if 'type' not in policy:
+        policy['type'] = InventoryRuleType.INVENTORY
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/operations/account.py:175]: May I ask why shall cli set a default type for policy? It seems more reasonable for service to set a default value for None properties.

File: src/storage-preview/azext_storage_preview/tests/latest/test_storage_account_scenarios.py [hunk 6]
```diff
@@ -120,3 +120,92 @@ def test_storage_account_resource_access_rules(self, resource_group, storage_acc
...
+                         JMESPathCheck("policy.rules[0].enabled", True),
+                         JMESPathCheck("policy.rules[0].name", "inventoryPolicyRule1"),
+                         JMESPathCheck("policy.type", "Inventory"),
+                         JMESPathCheck("resourceGroup", resource_group),
+                         JMESPathCheck("systemData", None)])
+
+        self.cmd('storage account blob-inventory-policy update --account-name {sa} -g {rg}'
+                 ' --set "policy.rules[0].name=newname"')
+        self.cmd('storage account blob-inventory-policy show --account-name {sa} -g {rg}',
+                 checks=JMESPathCheck('policy.rules[0].name', 'newname'))
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/tests/latest/test_storage_account_scenarios.py:208]: Could you check other properties didn't change except for rules[0].name changed?

File: src/storage-preview/azext_storage_preview/operations/account.py (hunk 5 above)
Comment (Juliehzl) [reply src/storage-preview/azext_storage_preview/operations/account.py:175]: at first, for the type property, there is only one value available. secondly, when there is no type in policy schema, service team don't want to break customer, so they would like to have a fallback when there is no type provided, which is also aligning with PSH.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**